Tests/js/views/dashboard/assets/ez-dashboardblockallcontentview-tests.js

```diff
@@ -0,0 +1,269 @@
+
+            view.template = function (params) {
+                templateCalled = true;
+
+                Y.Assert.isArray(params.items, 'The `items` variable should be an array');
+                Y.Assert.areSame(0, params.items.length, 'Should not provide any items data');
+                Y.Assert.isFalse(params.loadingError, 'The `loadingError` should not be enabled');
+
+                return origTpl.apply(this, arguments);
+            };
+
+            view.render();
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+            Y.Assert.isTrue(container.hasClass(CLASS_LOADING), 'Should add the loading state CSS class to the view container');
+        },
+
+        'Should render view with rows': function () {
+            var view = this.view,
+                contentMock = new Y.Mock(),
+                contentTypeMock = new Y.Mock(),
+                locationMock = new Y.Mock(),
+                contentInfoMock = new Y.Mock(),
+                origTpl = view.template,
+                templateCalled = false,
+                contentJSON = {},
+                contentTypeJSON = {},
+                contentInfoJSON = {},
+                locationJSON = {contentInfo: contentInfoJSON},
+                list;
+
+            Y.Mock.expect(contentMock, {
+                method: 'toJSON',
+                returns: contentJSON
+            });
+
+            Y.Mock.expect(contentTypeMock, {
+                method: 'toJSON',
+                returns: contentTypeJSON
+            });
+
+            Y.Mock.expect(contentInfoMock, {
+                method: 'toJSON',
+                returns: contentInfoJSON
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'toJSON',
+                returns: locationJSON
+            });
+
+            Y.Mock.expect(locationMock, {
+                method: 'get',
+                args: ['contentInfo'],
+                returns: contentInfoMock
+            });
+
+            list = [{
+                content: contentMock,
+                contentType: contentTypeMock,
+                location: locationMock
+            }];
+
+            view.template = function (params) {
+                templateCalled = true;
+
+                Y.Assert.isArray(params.items, 'The `items` variable should be an array');
+                Y.Assert.areSame(list.length, params.items.length, 'Should provide data of 1 item');
+                Y.Assert.areSame(contentJSON, params.items[0].content, 'Should provide content data of 1 item');
+                Y.Assert.areSame(contentTypeJSON, params.items[0].contentType, 'Should provide content type data of 1 item');
+                Y.Assert.areSame(locationJSON, params.items[0].location, 'Should provide location data of 1 item');
+                Y.Assert.areSame(contentInfoJSON, params.items[0].location.contentInfo, 'Should provide content info data of 1 item');
```

Comment:
this one is incorrect, in this context, the point is to check that the `contentInfo` is directly available without using the Location (basically the result of https://github.com/ezsystems/PlatformUIBundle/pull/589/files#diff-b26270100a3f8b2b43dbc1f0c0ed3604R70)

So this should be written:
```js
Y.Assert.areSame(contentInfoJSON, params.items[0].contentInfo, 'Should provide content info data of 1 item');
```